kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/BuildConfigTest.java

```diff
@@ -143,7 +143,7 @@ public void testBinaryBuildFromFile() throws IOException {
     File warFile = new File("target/test.war");
     warFile.createNewFile();
 
-    server.expect().post().withPath("/apis/build.openshift.io/v1/namespaces/ns1/buildconfigs/bc2/instantiatebinary?commit=&asFile=" + warFile.getName())
+    server.expect().post().withPath("/apis/build.openshift.io/v1/namespaces/ns1/buildconfigs/bc2/instantiatebinary?name=bc2&namespace=ns1&asFile=" + warFile.getName())
```

Comment:
Is there an option to parse the resource name `bc2` and namespace `ns1` from the request path itself? Why mention the same values again as query parameters when they are clearly provided in the path 
```
apis/{group}/{version}/namespaces/{namespaceName}/{resourceType}/{resourceName}/{subResource}
```

Reply:
I did this to align with `oc`'s behavior while doing binary builds. We don't have control over how these query parameters are processed since they are processed by openshift:

```
oc start-build random-generator-s2i --from-archive=target/docker/docker.io/rohankanojia/random-generator/0.0.1/tmp/docker-build.tar -v=8
I0710 19:23:02.785756   34958 loader.go:359] Config loaded from file /home/rohaan/.kube/config
I0710 19:23:02.786785   34958 loader.go:359] Config loaded from file /home/rohaan/.kube/config
I0710 19:23:02.787796   34958 loader.go:359] Config loaded from file /home/rohaan/.kube/config
I0710 19:23:02.788775   34958 loader.go:359] Config loaded from file /home/rohaan/.kube/config
I0710 19:23:02.789707   34958 loader.go:359] Config loaded from file /home/rohaan/.kube/config
Uploading archive file "target/docker/docker.io/rohankanojia/random-generator/0.0.1/tmp/docker-build.tar" as binary input for the build ...
I0710 19:23:02.789950   34958 round_trippers.go:383] POST https://api.rh-idev.openshift.com:443/apis/build.openshift.io/v1/namespaces/rokumar/buildconfigs/random-generator-s2i/instantiatebinary?name=random-generator-s2i&namespace=rokumar
I0710 19:23:02.789958   34958 round_trippers.go:390] Request Headers:
I0710 19:23:02.789964   34958 round_trippers.go:393]     Accept: application/json, */*
I0710 19:23:02.789971   34958 round_trippers.go:393]     User-Agent: oc/v1.11.0+d4cacc0 (linux/amd64) kubernetes/d4cacc0
I0710 19:23:02.789979   34958 round_trippers.go:393]     Authorization: Bearer tMfmFj52t-kLNdAKnpDgFJJk7OIlY4yUxqVBF_K8zjc
.I0710 19:23:08.404470   34958 round_trippers.go:408] Response Status: 201 Created in 5614 milliseconds
I0710 19:23:08.404496   34958 round_trippers.go:411] Response Headers:
I0710 19:23:08.404510   34958 round_trippers.go:414]     Content-Length: 1576
I0710 19:23:08.404529   34958 round_trippers.go:414]     Audit-Id: 930a4cf7-12f2-4865-b5e3-6055fcaca298
I0710 19:23:08.404544   34958 round_trippers.go:414]     Cache-Control: no-store
I0710 19:23:08.404569   34958 round_trippers.go:414]     Content-Type: application/json
I0710 19:23:08.404583   34958 round_trippers.go:414]     Date: Fri, 10 Jul 2020 13:53:08 GMT
I0710 19:23:08.404702   34958 request.go:897] Response Body: {"kind":"Build","apiVersion":"build.openshift.io/v1","metadata":{"name":"random-generator-s2i-4","namespace":"rokumar","selfLink":"/apis/build.openshift.io/v1/namespaces/rokumar/builds/random-generator-s2i-4","uid":"ad75af20-c2b4-11ea-9a28-0e6aaf341bbf","resourceVersion":"683137614","creationTimestamp":"2020-07-10T13:53:04Z","labels":{"app":"random-generator","buildconfig":"random-generator-s2i","group":"meetup","openshift.io/build-config.name":"random-generator-s2i","openshift.io/build.start-policy":"Serial","provider":"jkube","version":"0.0.1"},"annotations":{"openshift.io/build-config.name":"random-generator-s2i","openshift.io/build.number":"4","openshift.io/build.pod-name":"random-generator-s2i-4-build"},"ownerReferences":[{"apiVersion":"build.openshift.io/v1","kind":"BuildConfig","name":"random-generator-s2i","uid":"794bcaab-c1c2-11ea-af64-0e4452489d53","controller":true}]},"spec":{"serviceAccount":"builder","source":{"type":"Binary","binary":{}},"strategy":{"type":"Source","sourceStrategy":{"from":{"kin [truncated 552 chars]

Uploading finished
build.build.openshift.io/random-generator-s2i-4 started

```